JustSaying/Fluent/SubscriptionsBuilder.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using System.Collections.Generic;
+using JustSaying.Models;
+
+namespace JustSaying.Fluent
+{
+    /// <summary>
+    /// A class representing a builder for subscriptions. This class cannot be inherited.
+    /// </summary>
+    public sealed class SubscriptionsBuilder
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SubscriptionsBuilder"/> class.
+        /// </summary>
+        /// <param name="parent">The <see cref="MessagingBusBuilder"/> that owns this instance.</param>
+        internal SubscriptionsBuilder(MessagingBusBuilder parent)
+        {
+            Parent = parent;
+        }
+
+        /// <summary>
+        /// Gets the parent of this builder.
+        /// </summary>
+        internal MessagingBusBuilder Parent { get; }
+
+        /// <summary>
+        /// Gets the configured subscription builders.
+        /// </summary>
+        private IList<ISubscriptionBuilder<Message>> Subscriptions { get; } = new List<ISubscriptionBuilder<Message>>();
+
+        /// <summary>
+        /// Configures a queue subscription for the default queue.
+        /// </summary>
+        /// <returns>
+        /// The current <see cref="SubscriptionsBuilder"/>.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="configure"/> is <see langword="null"/>.
+        /// </exception>
+        public SubscriptionsBuilder ForQueue<T>()
+            where T : Message
+        {
+            return ForQueue<T>((p) => p.WithDefaultQueue());
+        }
+
+        /// <summary>
+        /// Configures a queue subscription.
+        /// </summary>
+        /// <param name="configure">A delegate to a method to use to configure a queue subscription.</param>
+        /// <returns>
+        /// The current <see cref="SubscriptionsBuilder"/>.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="configure"/> is <see langword="null"/>.
+        /// </exception>
+        public SubscriptionsBuilder ForQueue<T>(Action<QueueSubscriptionBuilder<T>> configure)
+            where T : Message
+        {
+            if (configure == null)
+            {
+                throw new ArgumentNullException(nameof(configure));
+            }
+
+            var builder = new QueueSubscriptionBuilder<T>();
+
+            configure(builder);
+
+            Subscriptions.Add(builder);
+
+            return this;
+        }
+
+        /// <summary>
+        /// Configures a topic subscription for the default topic name.
```

Comment:
I think we should explain this a bit more, given it will be the thing that most people will want to reach for.
Something that explains that you'll end up pulling from a queue with a subscription to the topic.

Reply:
We can beef-up the docs once we’ve committed to this new design, rather than write too much while it’s first-steps-y.